code/SecureFileExtension.php

```diff
@@ -92,6 +92,9 @@ public function canView($member = null) {
 		// File DataObjects created by SearchForm don't have a ParentID, which we need
 		// We fix this by re-getting the File object by it's ID if the ParentID is missing and use that
 		$file = $this->owner;
+		if( $file->ParentID == "" ){
+			$file = DataObject::get($file->ClassName)->byID($file->ID);
```

Comment:
On line #81 there's a `$this->owner->Parent()` which will fail before it gets to this line.